check_api/src/main/java/com/google/errorprone/VisitorState.java

```diff
@@ -406,20 +406,8 @@ public Name binaryNameFromClassname(String className) {
    */
   @Nullable
   public ClassSymbol getSymbolFromName(Name name) {
-    boolean modular = sharedState.modules.getDefaultModule() != getSymtab().noModule;
-    if (!modular) {
-      return getSymbolFromString(getSymtab().noModule, name);
-    }
-    for (ModuleSymbol msym : sharedState.modules.allModules()) {
-      ClassSymbol result = getSymbolFromString(msym, name);
-      if (result != null) {
-        // TODO(cushon): the path where we iterate over all modules is probably slow.
-        // Try to learn some lessons from JDK-8189747, and consider disallowing this case and
-        // requiring users to call the getSymbolFromString(ModuleSymbol, Name) overload instead.
-        return result;
-      }
-    }
-    return null;
+    ModuleSymbol msym = getSymtab().inferModule(Convert.packagePart(name));
+    return msym != null ? getSymbolFromString(msym, name) : null;
```

Comment:
[`Symtab#inferModule`](https://github.com/openjdk/jdk/blob/7cbf6721dcae6b0c22a57838d8dd769d87df9127/src/jdk.compiler/share/classes/com/sun/tools/javac/code/Symtab.java#L852-L871) performs a map lookup rather than a linear scan. This approach is derived from [this JDK logic](https://github.com/openjdk/jdk/blob/7cbf6721dcae6b0c22a57838d8dd769d87df9127/src/jdk.javadoc/share/classes/jdk/javadoc/internal/tool/ToolEnvironment.java#L158-L171). With it, the build still passes, as does the [Error Prone Support](https://github.com/PicnicSupermarket/error-prone-support/) build. That said, neither uses custom modularization, so I can't fully rule out that this logic behaves subtly different.